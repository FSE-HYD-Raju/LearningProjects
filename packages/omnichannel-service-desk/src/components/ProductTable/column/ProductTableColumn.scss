.ProductTableColumn {
	overflow: visible;
	text-overflow: ellipsis;
	margin-left: 0;
	margin-right: 0;
}

.ProductTableColumn-currency {
	color: #89ba17;
	font-weight: $font-weight-bold;
}

.ProductTableColumn-row {
	display: flex;
	flex: 1 1 auto;
	font-weight: 400;
}
