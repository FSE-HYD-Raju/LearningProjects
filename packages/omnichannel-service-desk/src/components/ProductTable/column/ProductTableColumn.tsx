import * as React from "react";
import { OcCurrency, ProductOffering, PriceRange, FormattedMessage } from "omnichannel-common-pos";
import messages from "../ProductTable.messages";
import { get } from "lodash";
import { TableColumn } from "./productTableColumns";

export interface ProductTableColumnProps {
	col: TableColumn;
	product: ProductOffering;
}

function PriceRenderer(range: PriceRange): React.ReactElement<void> {
	const min = range.min;
	const max = range.max;
	const currency = range.currency;

	if (max && max !== min) {
		return (
			<FormattedMessage
				{...messages.fromCurrency}
				values={{
					currency: (
						<OcCurrency
							cost={min}
							currency={currency}
							style={{
								color: "#89ba17",
								fontWeight: 500
							}}
						/>
					)
				}}
			/>
		);
	}
	if (min && currency) {
		return (
			<OcCurrency
				cost={min}
				currency={currency}
				className="ProductTableColumn-currency"
				replaceNonBreakingSpace={true}
			/>
		);
	}
	return (<span/>);
}

const ProductTableColumn: React.FC<ProductTableColumnProps> = (props: ProductTableColumnProps) => {
	const { col, product } = props;

	switch (col && col.type) {
		case "PRICE-RANGE": {
			const range: PriceRange = col.valueGetter ? col.valueGetter(product) : {};
			return (
				<div className="ProductTableColumn row">
					{PriceRenderer(range)}
				</div>
			);
		}

		default: {
			const columnValue =
				col && col.valueGetter
					? col.valueGetter(product)
					: get(product, col && col.attribute);

			return <span className="ProductTableColumn">{columnValue}</span>;
		}
	}
};

export default ProductTableColumn;
