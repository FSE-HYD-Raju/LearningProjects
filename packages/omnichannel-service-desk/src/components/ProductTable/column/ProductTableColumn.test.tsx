import * as React from "react";
import {
	ProductOfferingUtil,
	mountWithContext,
	shallowWithContext,
	ProductOffering
} from "omnichannel-common-pos";

import ProductTableColumn, { ProductTableColumnProps } from "./ProductTableColumn";

const product: ProductOffering = {
	id: "asdf"
} as ProductOffering;

const minProps: ProductTableColumnProps = {
	col: { type: "", flex: 1 },
	product: product
};

describe("ProductTableColumn", () => {
	it("succeeds at shallow mount with minimal props", () => {
		const wrapper = shallowWithContext(<ProductTableColumn {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimal props", () => {
		mountWithContext(<ProductTableColumn {...minProps} />);
	});

	it("is empty when currency is given but price is not", () => {
		const props: ProductTableColumnProps = {
			...minProps,
			col: { type: "PRICE-RANGE", flex: 1 }
		};

		const wrapper = mountWithContext(
			<ProductTableColumn {...props} />
		);

		expect(wrapper.text()).toEqual("");
	});

	it("renders product attribute correctly", () => {
		const props: ProductTableColumnProps = {
			col: {
				...minProps.col,
				attribute: "name"
			},
			product: {
				...minProps.product,
				name: "VIP Product"
			}
		};
		const wrapper = mountWithContext(
			<ProductTableColumn {...props} />
		);

		expect(wrapper.find(".ProductTableColumn").text()).toContain("VIP Product");
	});

	it("renders price range", () => {
		const product: ProductOffering = {
			id: "asdf",
			attributes: {
				prices: [
					{
						taxFreeAmount: 66.6,
						currency: "EUR",
						type: "ONE_TIME"
					},
					{
						taxFreeAmount: 99.9,
						currency: "USD",
						type: "ONE_TIME"
					}
				]
			}
		} as ProductOffering;
		const props: ProductTableColumnProps = {
			col: {
				...minProps.col,
				type: "PRICE-RANGE",
				attribute: "Zamzung Qalaxy",
				valueGetter: ProductOfferingUtil.getUpfrontPriceRange
			},
			product: product
		};

		const wrapper = mountWithContext(
			<ProductTableColumn {...props} />
		);

		const productTableColumn = wrapper
			.find(".ProductTableColumn")
			.hostNodes();
		expect(productTableColumn.text()).toContain("from");
		expect(productTableColumn.text()).toContain("66");
		expect(productTableColumn.text()).toContain("60");
	});
});
