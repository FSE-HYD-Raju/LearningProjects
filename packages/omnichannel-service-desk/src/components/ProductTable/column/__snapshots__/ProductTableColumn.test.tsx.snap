// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTableColumn succeeds at shallow mount with minimal props 1`] = `
<span
  className="ProductTableColumn"
/>
`;
